scripts/goal_agent_eval_matrix.sh

```
@@ -0,0 +1,78 @@
+
+if [[ -z "${GOAL_AGENT_EVAL_API_KEY:-${MELIOUS_API_KEY:-}}" ]]; then
+  echo "Set MELIOUS_API_KEY (or GOAL_AGENT_EVAL_API_KEY)." >&2
+  exit 2
+fi
+
+mkdir -p "$LOG_DIR"
+
+# Warm the build once. Otherwise every process in the first wave compiles the
+# same test concurrently, which is slower than compiling it once.
+echo "Warming the test build..."
+LOTTI_GOAL_AGENT_EVAL_LIVE= fvm flutter test "$TEST_PATH" >/dev/null 2>&1
```

Shellcheck (0.11.0)

[warning] Remove space after = if trying to assign a value (for empty string, use var='' ... ).

(SC1007)